lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php

```diff
@@ -260,14 +299,16 @@
     /**
      * READ-ONLY: The array of indexes for the document collection.
      *
-     * @var array
+     * @var array<array<string, mixed>>
+     * @psalm-var array<array{keys: array<string, mixed>, options: array<string, mixed>}>
```

Comment:
I'm wondering if we should define types for this. Consider:
```suggestion
     * @psalm-var array<IndexMapping>
```

Defining an `IndexMapping` psalm type elsewhere that makes use of other types (e.g. `IndexKeys` and `IndexOptions`) would make for easier reading and understanding of what is actually stored.

Reply:
Good idea! Done, I've defined them in `ClassMetadata`, but I guess maybe should belong somewhere else.